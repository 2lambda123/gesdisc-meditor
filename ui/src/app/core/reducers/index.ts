import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
