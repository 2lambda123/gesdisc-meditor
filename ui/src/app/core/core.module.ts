import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
