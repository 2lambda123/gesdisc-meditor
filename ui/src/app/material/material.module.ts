import { NgModule } from '@angular/core';

import {
  MatInputModule,
  MatCardModule,
  MatButtonModule,
  MatSidenavModule,
  MatListModule,
  MatIconModule,
  MatToolbarModule,
  MatProgressSpinnerModule,
  MatDatepickerModule,
  MatCheckboxModule,
  MatSelectModule
} from '@angular/material';

@NgModule({
  imports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
  exports: [
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatSidenavModule,
    MatListModule,
    MatIconModule,
    MatToolbarModule,
    MatProgressSpinnerModule,
    MatDatepickerModule,
    MatCheckboxModule,
  	MatSelectModule
  ],
})
export class MaterialModule {}
