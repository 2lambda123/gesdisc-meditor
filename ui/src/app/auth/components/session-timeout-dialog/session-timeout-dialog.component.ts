import { Component } from '@angular/core';
import { MatDialogRef } from '@angular/material';

import { environment } from '../../../../environments/environment';

@Component({
	selector: 'med-session-timeout-dialog',
	template: `
		<mat-dialog-content>
			<h3>Your session has expired, please login again</h3>
			<p>If you were working on something, be sure to close this dialog and copy your changes out.</p>
			<p>Unsaved changes will be lost!</p>
        
			<button mat-button (click)="login()" color="accent" class="login-btn">
				<mat-icon>person</mat-icon>
				Login
			</button>
        </mat-dialog-content>`,
	styles: [
		`
			h1, h5, mat-dialog-content {
				text-align: center;
			}

			h1 {
				font-size: 24px;
			}

			.login-btn {
				margin: 10px 0;
			}
		`
	]
})

export class SessionTimeoutDialog {

	constructor(public dialogRef: MatDialogRef<SessionTimeoutDialog>) {}

	login() {
		window.location.href = this.getApiUrl() + '/login';
	}

	getApiUrl() {
		const basePath = environment.API_BASE_PATH;
		return basePath.indexOf('http') !== 0 ? window.location.origin + basePath : basePath;
	}

}
