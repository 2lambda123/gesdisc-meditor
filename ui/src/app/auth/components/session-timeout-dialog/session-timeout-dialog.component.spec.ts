import { TestBed, ComponentFixture } from '@angular/core/testing';
import { MatIconModule, MatDialogModule, MatDialogRef } from '@angular/material';
import { SessionTimeoutDialog } from './session-timeout-dialog.component';

describe('SessionTimeoutDialog Component', () => {
	let fixture: ComponentFixture<SessionTimeoutDialog>;
	let component: SessionTimeoutDialog;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				MatIconModule,
				MatDialogModule
			],
			declarations: [ SessionTimeoutDialog ],
			providers: [
				{ provide: MatDialogRef, useValue: {} }
			]
		});

		fixture = TestBed.createComponent(SessionTimeoutDialog);
		component = fixture.componentInstance;

		fixture.detectChanges();
	});

	it('should compile session timeout dialog', () => {
		expect(fixture).toMatchSnapshot();
	});
});
