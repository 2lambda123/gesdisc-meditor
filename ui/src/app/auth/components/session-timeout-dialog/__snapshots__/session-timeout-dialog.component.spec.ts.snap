// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionTimeoutDialog Component should compile session timeout dialog 1`] = `
<med-session-timeout-dialog
  dialogRef={[Function Object]}
>
  <mat-dialog-content
    _ngcontent-c0=""
    class="mat-dialog-content"
  >
    <h3
      _ngcontent-c0=""
    >
      Your session has expired, please login again
    </h3>
    <p
      _ngcontent-c0=""
    >
      If you were working on something, be sure to close this dialog and copy your changes out.
    </p>
    <p
      _ngcontent-c0=""
    >
      Unsaved changes will be lost!
    </p>
    <button
      _ngcontent-c0=""
      class="login-btn"
      color="accent"
      mat-button=""
    >
      <mat-icon
        _ngcontent-c0=""
        aria-hidden="true"
        class="mat-icon material-icons mat-icon-no-color"
        role="img"
      >
        person
      </mat-icon>
       Login 
    </button>
  </mat-dialog-content>
</med-session-timeout-dialog>
`;
